packages/dds/tree/src/shared-tree/treeCheckout.ts

```diff
@@ -559,6 +590,28 @@ export class TreeCheckout implements ITreeCheckoutFork {
 		}
 	};
 
+	/**
+	 * Applies the given serialized change (as was produced via a `"changed"` event of another checkout) to this checkout.
+	 */
+	public applySerializedChange(serializedChange: JsonCompatibleReadOnly): void {
+		if (!isSerializedChange(serializedChange)) {
+			throw new UsageError(`Cannot apply change. Invalid serialized change format.`);
+		}
+		const { revision, parent, change } = serializedChange;
+		if (parent !== this.#transaction.activeBranch.getHead().revision) {
+			throw new UsageError(
+				`Cannot apply change. A serialized changed must be applied to the same branch state as it was created from.`,
+			);
+		}
+		const context: ChangeEncodingContext = {
+			idCompressor: this.idCompressor,
+			originatorId: this.idCompressor.localSessionId,
+			revision,
+		};
+		const decodedChange = this.changeFamily.codecs.resolve(4).json.decode(change, context);
+		this.applyChange(decodedChange, revision);
```

Comment:
Don't we need to do this before applying the change? (quoting our Teams conversation)

> We do have a way that we can allow the sandbox ID compressor to have a different base UUID (session ID) than the outer compressor. After each edit, or after a batch of edits (but before applying them to the outer view), we get the range of IDs that were generated by the sandbox. We then sequence that range by submitting and waiting for the round-trip of an IdCompressor op containing that range. Now all clients (including our local client) are able to safely use any of those IDs.

Or are we only currently supporting applying changes from the same tree checkout?